### DISCUSSION TITLE
Есть ли необходимость в двух профилях для Organisation?

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#124] [Административная РГ]
C учетом того что в Organisation.Type мы требуем указывать типизацию Организация/Подразделение (решение принято на встрече http://fhir-ru.zendoc.me/meetings.2022-04-05), есть ли необходимость в различных профилях для них?

Решение 1: делаем два профиля

Плюсы: 
Удобно, что в профиле юридического лица можно указать свои примеры классификации, определить PartOf (0..0), а в профиле подразделения - PartOf (1..1).

Минусы: 
Информация о враче - авторе документа, назначения и т.п. передается в виде «Петров П.П., врач-терапевт 2 терапевтического отделения», или в виде «Петров П.П., врач-терапевт Медицинского центра ABC». Информация об организации здесь кодируется атрибутом PractitionerRole.Organisation. Если мы делаем два профиля, то не сможем указать однозначно профиль для этого атрибута.

Решение 2: делаем один профиль, кардинальность описываем слайсами

**Comment by eugenekogan-spb** [comment]
Из обсуждения 2022 04 08 адм: надо уточнить, возможно ли средствами профиля (StructureDefinition) указать, что при одном значении Type будут одни кардинальности, а при другом - другие. Если да, то можно оставить один профиль.

**Reply by Vrodvi** [reply]
Различие в кардинальности говорит о том что описываются различные структуры, а значит проще создать отдельные описания под каждую структуру - т.е. отдельные профили\StructureDefinition для организации и подразделения.

Нужно помнить, что создаются корневые профили и сложные (определяемые через условие) правила кардинальности будут наследоваться во все последующие профили из-за чего профили могут стать сложными для понимания с большим количество вариабельных элементов.
В то время как отдельные профили будут сосредоточены на одно конкретной простой структуре.
Поэтому я за вариант с отдельными профилями.

Также прошу рассмотреть подход с тремя профилями:
- (1) **базовый профиль** (с переводами всех описаний на русский и с минимальными ограничениями)
  - (2) наследуемый от корневого профиля **профиль организации**
  - (3) наследуемый от корневого профиля **профиль подразделения**

--- 

И ещё предостережение - в базовых профилях желательно накладывать минимальные ограничения, и затем ограничивать их в специализированных профилях, иначе мы получим слишком жёсткую структуру в которую невозможно будет поместить реальные данные.
Например, ограничение в PartOf (0..0) для организаций выглядит слишком радикальным вариантом - не понятно как тогда можно будет обозначить филиалы юридического лица, которые являются его составной частью (т.е. как раз "part of'")?

**Comment by eugenekogan-spb** [comment]
К обсуждению на встрече 2022-04-12 - и в комментарии выше:
@Vrodvi предложил считать филиал вариантом юридического лица и ставить PartOf на основное юрлицо.
Однако, по закону от 8 августа 2001 г. N 129-ФЗ "О государственной регистрации юридических лиц и индивидуальных предпринимателей", Статья 5, п.1 н) - https://base.garant.ru/12123875/5633a92d35b966c2ba2f1e859e7bdd69/#block_501 - получается, что сведения о филиале входят в регистрационную карточку юрлица. То есть, филиал сам юрлицом не является. Он является вариантом подразделения, просто пафосно названным и обязательно (по закону) находящийся по другому адресу - см. например https://www.garant.ru/consult/civil_law/1432767/

**Reply by Vrodvi** [reply]
Я и не утверждал, что филиал является юрлицом, а просто указал на то что филиалы юрлица описаны в реестре ЕГРЮЛ вместе с информацией о самом юрлице (в отличии от отделений\подразделений, которых в карточке юрлица не указано).

Обсуждалось, что нужен некий формальный критерий, по которому можно будет отделить что является организацией, а что подразделением. Таким критерием может быть информация из карточки юрилица - само юрлицо и его филиалы (указанные в карточке юрлица) можно считать "организациями", а все остальные структурные элементы (как выше "организации" - например, объединения организаций, так и ниже - отделения, кабинеты и прочее) уже считаем "подразделением".

Кроме того, филиал - не просто _пафосно названное подразделение_, а полноценная единица налогового учёта, что видно из карточки юрлица - для филиалов присваиваются налоговые атрибуты, например [КПП](https://www.banki.ru/wikibank/kod_prichinyi_postanovki_na_uchet/) - например для юрлица **ФГБУ "НМИЦ РАДИОЛОГИИ" МИНЗДРАВА РОССИИ** (ОГРН 1 0 2 7 7 3 9 6 2 3 0 3 1):
- налоговая информация самого юрлица:
30 ИНН юридического лица 7714042070
31 КПП юридического лица 402501001
- налоговая информация филиалов:
200 КПП филиала: 771443001
208 КПП филиала: 402543001
218 КПП филиала: 771943001
см. [ul-1027739623031-20220413093253.pdf](https://github.com/fhir-ru/core/files/8478822/ul-1027739623031-20220413093253.pdf) (https://egrul.nalog.ru/index.html)
Если будем считать филиалы просто подразделениями, то у подразделений придётся создавать атрибут для указания КПП, но не все подразделения будут его иметь.

Т.е. филиалы всё же отличаются от всех остальных структурных подразделений, как минимум по двум критериям - они прописаны как составная часть юрлица и они являются единицами налогового учёта.
Считаю что это может служить достаточным формальным критерием разделения организаций от подразделений.

Если имеются иные мнения, то просьба приводить их с указанием альтернативных формальных критериев, по которым можно осуществлять деление организация\подразделение.
